google-cloud-examples/src/main/java/com/google/cloud/examples/bigtable/TableAdminExample.java

```diff
@@ -0,0 +1,373 @@
+  }
+
+  /** Demonstrates how to list all tables within an instance. */
+  public void listAllTables() {
+    System.out.println("\nListing tables in current instance");
+    // [START bigtable_list_tables]
+    // Lists tables in the current instance.
+    try {
+      List<String> listTables = adminClient.listTables();
+      for (String tableName : listTables) {
+        System.out.println(tableName);
+      }
+    } catch (NotFoundException e) {
+      System.err.println("Failed to list tables from a non-existent instance: " + e.getMessage());
+    }
+    // [END bigtable_list_tables]
+  }
+
+  /** Demonstrates how to get a table's metadata. */
+  public void getTableMeta() {
+    System.out.println("\nPrinting table metadata");
+    // [START bigtable_get_table_metadata]
+    // Gets table metadata, and applies a view to the table fields.
+    try {
+      Table table = adminClient.getTable(tableId);
+      System.out.println("Table: " + table.getId());
+      Collection<ColumnFamily> columnFamilies = table.getColumnFamilies();
+      for (ColumnFamily columnFamily : columnFamilies) {
+        printColumnFamily(columnFamily);
+      }
+    } catch (NotFoundException e) {
+      System.err.println(
+          "Failed to retrieve table metadata for a non-existent table: " + e.getMessage());
+    }
+    // [END bigtable_get_table_metadata]
+  }
+
+  /** Demonstrates how to create a new instance of the DurationRule. */
+  public void maxAgeRule() {
+    System.out.printf("%nCreating column family %s with max age GC rule%n", COLUMN_FAMILY_1);
+    // [START bigtable_create_family_gc_max_age]
+    // Creates a column family with GC policy : maximum age
+    // where age = current time minus cell timestamp
+
+    // Defines the GC rule to retain data with max age of 5 days.
+    GCRules.DurationRule maxAgeRule1 = GCRules.GCRULES.maxAge(5, TimeUnit.DAYS);
+
+    // Creates column family with given GC rule.
+    try {
+      ModifyColumnFamiliesRequest columnFamiliesRequest1 =
+          ModifyColumnFamiliesRequest.of(tableId).addFamily(COLUMN_FAMILY_1, maxAgeRule1);
+      adminClient.modifyFamilies(columnFamiliesRequest1);
+      System.out.println("Created column family: " + COLUMN_FAMILY_1);
+    } catch (AlreadyExistsException e) {
+      System.err.println(
+          "Failed to create column family with rule, already exists: " + e.getMessage());
+    }
+    // [END bigtable_create_family_gc_max_age]
+  }
+
+  /** Demonstrates how to create a new instance of the VersionRule. */
+  public void maxVersionsRule() {
+    System.out.printf("%nCreating column family %s with max versions GC rule%n", COLUMN_FAMILY_2);
+    // [START bigtable_create_family_gc_max_versions]
+    // Creates a column family with GC policy : most recent N versions
+    // where 1 = most recent version
+
+    // Defines the GC policy to retain only the most recent 2 versions.
+    GCRules.VersionRule versionRule1 = GCRules.GCRULES.maxVersions(2);
```

Comment:
The variable names are confusing, why is versionRule1 used for columnFamiliesRequest2?
If I understand correctly, the docs will only contain the snippet between `[START bigtable_create_family_gc_max_versions]` and `[END bigtable_create_family_gc_max_versions]`, so the reader will have very little context about the overall naming scheme. I would try to keep the names meaningful within the snippet.

Reply:
Also, please use a static import for `GCRULES`